v2/spanner-common/src/main/java/com/google/cloud/teleport/v2/spanner/migrations/shard/CassandraShard.java

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright (C) 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.teleport.v2.spanner.migrations.shard;
+
+import com.datastax.oss.driver.api.core.config.DefaultDriverOption;
+import com.datastax.oss.driver.api.core.config.DriverConfigLoader;
+import com.datastax.oss.driver.api.core.config.DriverExecutionProfile;
+import java.util.List;
+import java.util.Objects;
+
+public class CassandraShard extends Shard {
+  private final DriverConfigLoader configLoader;
```

Comment:
Please use OptionsMap because CassandraShard needs to be Serializable

Reply:
@shreyakhajanchi can you elaborate more about this, I guess I am not sure we need serialized there

Reply:
Shard class implements Serializable and is being used as a class member in the pTransform functions which implements Serializable
cc: @VardhanThigle

Reply:
@shreyakhajanchi agree and adress the same